phoenix5-spark/src/main/java/org/apache/phoenix/spark/datasource/v2/PhoenixDataSource.java

```diff
@@ -33,21 +37,27 @@
 import org.apache.spark.sql.sources.v2.reader.DataSourceReader;
 import org.apache.spark.sql.sources.v2.writer.DataSourceWriter;
 import org.apache.spark.sql.types.StructType;
+import scala.collection.immutable.Map;
 
 import static org.apache.phoenix.util.PhoenixRuntime.JDBC_PROTOCOL;
 import static org.apache.phoenix.util.PhoenixRuntime.JDBC_PROTOCOL_SEPARATOR;
 
 /**
  * Implements the DataSourceV2 api to read and write from Phoenix tables
  */
-public class PhoenixDataSource  implements DataSourceV2,  ReadSupport, WriteSupport, DataSourceRegister {
+public class PhoenixDataSource implements DataSourceV2, ReadSupport, WriteSupport, DataSourceRegister, RelationProvider {
 
     private static final Logger logger = LoggerFactory.getLogger(PhoenixDataSource.class);
     public static final String SKIP_NORMALIZING_IDENTIFIER = "skipNormalizingIdentifier";
     @Deprecated
     public static final String ZOOKEEPER_URL = "zkUrl";
     public static final String JDBC_URL = "jdbcUrl";
     public static final String PHOENIX_CONFIGS = "phoenixconfigs";
+    public static final String TABLE = "table";
+    public static final String DATE_AS_TIME_STAMP = "dateAsTimestamp";
+    public static final String DO_NOT_MAP_COLUMN_FAMILY = "doNotMapColumnFamily";
+    public static final String TENANT_ID = "tenantId";
```

Comment:
use PhoenixConfigurationUtil.MAPREDUCE_TENANT_ID as value instead of repeating the string ?

Reply:
I think PhoenixConfigurationUtil.MAPREDUCE_TENANT_ID have another usage and can't be used here.
The TENANT_ID property is only used at spark options level. For setting the value for phoenix connection property  I use PhoenixRuntime.TENANT_ID_ATTRIB. 
I prefer keep them separate and duplicate value  because PhoenixRuntime.TENANT_ID_ATTRIB come from a phoenix dependency so if its value changes it will have an effect on spark connector.